### PR TITLE
Quarterly SIG Release teams update

### PR DESCRIPTION
**Pull request by justaugustus**
- Add ~SIG Release Chairs~ Release Engineering subproject owners as maintainers for CIP
- Promote cpanato to Branch Manager
- Add ps882 to Build Admins
- Remove inactive Release Manager Associates
- Update `milestone-maintainers` (compared w/ current SIG Chairs/TLs)
- Remove ixdy from `release-engineering`
- Remove 1.15 team from `release-team`
- Rotate `release-team-leads`
- Update `sig-release` GH team

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

(This also syncs up `OWNERS_ALIASES` w/ k/community)

/hold
/sig release
/area release-eng release-team
/assign @tpepper @calebamiles @mrbobbytables @nikhita @cblecker 
cc: @kubernetes/release-engineering @kubernetes/release-team @kubernetes/milestone-maintainers

### CODE REVIEW

**Comment by justaugustus** [comment]
@cpanato -- hang on. There are some errors to fix in https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/org/1205/pull-org-test-all/1175699913594376198

**Comment by justaugustus** [comment]
Hmmm, I wonder if [this new failure](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/org/1205/pull-org-test-all/1175703815580553217) is transient...
```bazel
INFO: Call stack for the definition of repository 'zlib' which is a http_archive (rule definition at /root/.cache/bazel/_bazel_root/e50ebc2a7cac8d93968e5988863c67ce/external/bazel_tools/tools/build_defs/repo/http.bzl:237:16):
 - /root/.cache/bazel/_bazel_root/e50ebc2a7cac8d93968e5988863c67ce/external/com_google_protobuf/protobuf_deps.bzl:9:9
 - /home/prow/go/src/github.com/kubernetes/org/WORKSPACE:102:1
ERROR: An error occurred during the fetch of repository 'zlib':
   java.io.IOException: Error downloading [https://zlib.net/zlib-1.2.11.tar.gz] to /root/.cache/bazel/_bazel_root/e50ebc2a7cac8d93968e5988863c67ce/external/zlib/zlib-1.2.11.tar.gz: Unknown host: zlib.net
ERROR: /root/.cache/bazel/_bazel_root/e50ebc2a7cac8d93968e5988863c67ce/external/com_google_protobuf/BUILD:150:1: @com_google_protobuf//:protobuf depends on @zlib//:zlib in repository @zlib which failed to fetch. no such package '@zlib//': java.io.IOException: Error downloading [https://zlib.net/zlib-1.2.11.tar.gz] to /root/.cache/bazel/_bazel_root/e50ebc2a7cac8d93968e5988863c67ce/external/zlib/zlib-1.2.11.tar.gz: Unknown host: zlib.net
ERROR: Analysis of target '//admin:update' failed; build aborted: no such package '@zlib//': java.io.IOException: Error downloading [https://zlib.net/zlib-1.2.11.tar.gz] to /root/.cache/bazel/_bazel_root/e50ebc2a7cac8d93968e5988863c67ce/external/zlib/zlib-1.2.11.tar.gz: Unknown host: zlib.net
```

/test pull-org-test-all

File: config/kubernetes-sigs/sig-release/teams.yaml [hunk 1]
```diff
@@ -1,17 +1,21 @@
 teams:
   k8s-container-image-promoter-admins:
-    description: "admin access to k8s-container-image-promoter"
+    description: admin access to k8s-container-image-promoter
```
Comment (cblecker) [inline config/kubernetes-sigs/sig-release/teams.yaml]: One comment: While it is up to sig-release to administer their repos as they see fit, I think it's an anti-pattern to tie sig chairpersonship to a repos permission set.

Chairs are a process role, as defined in the dig governance docs. There are certain places where it makes sense for the chairs/leads to automatically be in owners files (mainly k/community meta docs, and k/org github meta data -- specifically where being in the owners file is directly tied to administering the sig), but I think in most cases actual access to things should be delegated to subproject owners.

This is similar to how the steering committee no longer gets automatic github owner to everything, rather they delegate that to the subproject that administers it.

To be clear though, this is just my opinion, and is not blocking (strictly speaking)

cc: @kubernetes/sig-release

Comment (justaugustus) [reply config/kubernetes-sigs/sig-release/teams.yaml]: @cblecker -- That's fair. 

To be more explicit, SIG Release Chairs are [subproject owners](https://github.com/kubernetes/sig-release/blob/cdbd4045aded4a6c45ec81b308f7f8de07ec5eee/OWNERS_ALIASES#L11-L19) of the Release Engineering project and [should have ownership of all tools/processes which have the potential to affect a Kubernetes release](https://github.com/kubernetes/sig-release/issues/732).

I was going to cite some things from our charter, specifically, [deviations from SIG governance](https://github.com/kubernetes/community/blob/master/sig-release/charter.md#deviations-from-sig-governance) e.g., SIG Release does not have Technical Leads, as the Chairs are required to be technical, but it seems the charter is [due for some updates](https://github.com/kubernetes/sig-release/issues/378) (which I'll action on this cycle).

p.s. agreed on the Steering point; I did open [the issue](https://github.com/kubernetes/org/issues/637), after all. :stuck_out_tongue_closed_eyes:

File: config/kubernetes/sig-release/teams.yaml [hunk 2]
```diff
@@ -27,103 +28,97 @@ teams:
...
     - fejta # Testing
-    - idvoretskyi # PM
-    - mrbobbytables # 1.15 Enhancements Shadow
+    - mrbobbytables # 1.17 Enhancements Lead
     - nikhita # ContribEx
     - spiffxp # Testing
     members:
     - abgworrall # GCP
-    - alejandrox1 # 1.15 CI Signal
+    - adisky # OpenStack
```
Comment (nikhita) [inline config/kubernetes/sig-release/teams.yaml]: Is there any particular reason we are still providing access to old SIG-<cloud-provider>-leads now that they've been folded into SIG Cloud Provider?

File: config/kubernetes/sig-release/teams.yaml [hunk 3]
```diff
@@ -246,32 +233,39 @@ teams:
...
+        - alejandrox1 # 1.17 RT Lead Shadow
+        - guineveresaenger # 1.17 RT Lead
+        - jeefy # 1.17 RT Lead Shadow
+        - mariantalla # 1.17 RT Lead Shadow
         privacy: closed
   sig-release:
     description: SIG Release Members
     maintainers:
     - mrbobbytables
+    - nikhita
```
Comment (nikhita) [inline config/kubernetes/sig-release/teams.yaml]: Out of curiosity, how do we decide who gets added to this team? Is it members who have previously participated in some form in the release team? Is this team also pruned periodically/after a release?

File: config/kubernetes/sig-release/teams.yaml (hunk 2 above)
Comment (justaugustus) [reply config/kubernetes/sig-release/teams.yaml]: @nikhita -- there's still provider code in k/k, so those contributors will likely still need milestone maintainers privs.

File: config/kubernetes/sig-release/teams.yaml (hunk 3 above)
Comment (justaugustus) [reply config/kubernetes/sig-release/teams.yaml]: @nikhita -- SIG Release has the concept of membership --> https://github.com/kubernetes/community/blob/master/sig-release/charter.md#sig-membership

tl;dr Chairs, subproject owners, former RT role leads

We don't currently prune anyone; only add, but members can definitely ask to be removed.

File: config/kubernetes-sigs/sig-release/teams.yaml (hunk 1 above)
Comment (cblecker) [reply config/kubernetes-sigs/sig-release/teams.yaml]: Ahh okay. I think we then just need to be explicit that Stephen and Caleb are added to this group because they are subproject owners, *not* because they are chairs. They're two distinct jobs, even though the same person can hold both. Is that right?

Comment (justaugustus) [reply config/kubernetes-sigs/sig-release/teams.yaml]: @cblecker -- yep, yep. Updated the PR description and commit comment to reflect.

**Comment by justaugustus** [comment]
@tpepper @calebamiles @cblecker @nikhita -- this is ready for approval.

**Comment by timothysc** [comment]
LGTM 

But I'd also like SIGs to maybe cross ref this and take ownership/responsibility for updating their owners files.

**Comment by cblecker** [comment]
/approve
/hold

Approving for root. Explicit hold so this doesn't merge before @justaugustus is ready for it to.
